### ISSUE TITLE
always generate email token in registration, even if "email" has not sent

### ISSUE THREAD

**Issue by yariksav** [adonisjs/adonis-persona#11]
What if user registration doing without "email", for example one of "uid" is phone number or username?
This code
https://github.com/adonisjs/adonis-persona/blob/dd573db6f328ef8c910b5dfdc40e751cfccc90a1/src/Persona.js#L491
always generate email token, even if email was not send

Perhaps need to add condition if email has sent?

**Comment by thetutlage** [comment]
Yeah, we can improve it by introducing another config property called `verificationTokenField`. 

1. If this field is empty, no token will be generated
2. If this field has value, a token for that value will be generated. You can use it to send email and as you mentioned, you can use it to send an SMS to the phone.

**Comment by yariksav** [comment]
Hi @thetutlage!
> If this field is empty, no token will be generated

Great idea! If some projects will have only username/password and work without email or phone

> You can use it to send email and as you mentioned, you can use it to send an SMS to the phone.

But what if user can choose method of registration, by phone or by email address, or for two-factor authorization for instance.
In this logic we have to be able to send verification to both, or to one of them...
🤔
